1st chunk: XXX pkg-config(1) bug; prevent failure in configuring gnome-shell:
Package gsettings-desktop-schemas>=47.beta was not found in the pkg-config search path

XXX push upstream

Index: meson.build
--- meson.build.orig
+++ meson.build
@@ -26,7 +26,7 @@ pango_req = '>= 1.46.0'
 cairo_req = '>= 1.10.0'
 pangocairo_req = '>= 1.20'
 pixman_req = '>= 0.42'
-gsettings_desktop_schemas_req = '>= 47.beta'
+gsettings_desktop_schemas_req = '>= 47'
 x11_req = '>= 1.7.0'
 xcomposite_req = '>= 0.4'
 xkbcommon_req = '>= 0.4.3'
@@ -596,6 +596,8 @@ if have_xwayland
     cdata.set('HAVE_SYS_RANDOM', 1)
   elif cc.has_header_symbol('linux/random.h', 'getrandom')
     cdata.set('HAVE_LINUX_RANDOM', 1)
+  elif cc.has_header_symbol('stdlib.h', 'arc4random_buf')
+    cdata.set('HAVE_ARC4RANDOM_BUF', 1)
   else
     error('Required function getrandom not found')
   endif
